dotcom-rendering/scripts/env/check-deps.js

```diff
@@ -9,25 +8,30 @@ if (pkg.devDependencies) {
 	process.exit(1);
 }
 
-const { object: json } = lockfile.parse(
-	fs.readFileSync('../yarn.lock', 'utf8'),
-);
-
 const knownNonSemver = /** @type {const} */ ([
 	'https://github.com/guardian/babel-plugin-px-to-rem#v0.1.0',
 ]);
 
 const mismatches = Object.entries(pkg.dependencies)
-	.filter(([name, version]) => {
-		const pinned = json[name + '@' + version]?.version;
-		return version !== pinned;
-	})
+	.filter(
+		([name, version]) =>
+			version.includes('*') ||
+			version.includes('x') ||
+			version.includes(' - ') ||
+			version.startsWith('^') ||
+			version.startsWith('~') ||
+			version.startsWith('>') ||
+			version.startsWith('<'),
+	)
```

Comment:
> **Note**
> this does the opposite of what we did previsouly, which was to filter out valid deps

Given that what we really want is fully pinned dependencies, I think that stricly allowing `MAJOR.MINOR.PATCH` via a regex would be more explicit and less error prone:

```suggestion
	.filter(
		([, version]) =>
			!version.match(/^(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)$/),
	)
```

The `package.json` is very liberal in what it can accept as a valid version and this is precisely what we’re trying to prevent. For example, a version defined exclusively with a major version should be invalid.

Reply:
what if the dep is a [pre-release](https://semver.org/#spec-item-9) or contains meta info? both could be possible for an internal package

Reply:
I think that in that case it should be handled explicitely using the existing `knownNonSemver` array, which we could rename to `knownDeviations` or something more generic. An allowlist has the added benefit that comments can be added to explain why a specific dependency is specified a certain way. We can update the warning to suggest this, too, if we find it too restrictive.

> Dependency X must match MAJOR.MINOR.PATCH version. You may add it to a list of known deviations if it cannot meet this format.